src/main/java/org/jabref/gui/actions/ExportLinkedFilesAction.java

```diff
@@ -0,0 +1,143 @@
+
+import javax.swing.AbstractAction;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.concurrent.Service;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Dialog;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TableCell;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.paint.Color;
+import javafx.scene.text.Text;
+
+import org.jabref.Globals;
+import org.jabref.JabRefGUI;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.FXDialogService;
+import org.jabref.gui.util.DefaultTaskExecutor;
+import org.jabref.gui.util.DirectoryDialogConfiguration;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.preferences.JabRefPreferences;
+
+import de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIcon;
+import de.jensd.fx.glyphs.materialdesignicons.utils.MaterialDesignIconFactory;
+
+public class ExportLinkedFilesAction extends AbstractAction {
+
+    private final DialogService dialogService = new FXDialogService();
+    private BibDatabaseContext databaseContext;
+    private List<BibEntry> entries;
+
+    public ExportLinkedFilesAction() {
+        super(Localization.lang("Copy attached files to folder..."));
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+
+        DirectoryDialogConfiguration dirDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                .withInitialDirectory(Paths.get(Globals.prefs.get(JabRefPreferences.EXPORT_WORKING_DIRECTORY)))
+                .build();
+        entries = JabRefGUI.getMainFrame().getCurrentBasePanel().getSelectedEntries();
+
+        Optional<Path> exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> dialogService.showDirectorySelectionDialog(dirDialogConfiguration));
+
+        exportPath.ifPresent(path -> {
+            databaseContext = JabRefGUI.getMainFrame().getCurrentBasePanel().getDatabaseContext();
+
+            Service<List<CopyFilesResultViewModel>> exportService = new ExportLinkedFilesService(databaseContext, entries, path);
+            startServiceAndshowProgessDialog(exportService);
+        });
+    }
+
+    private void startServiceAndshowProgessDialog(Service<List<CopyFilesResultViewModel>> exportService) {
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+            exportService.setOnSucceeded(value -> {
+                DefaultTaskExecutor.runInJavaFXThread(() -> showDialog(exportService.getValue()));
+            });
+            exportService.start();
+            DialogService dlgService = new FXDialogService();
+            dlgService.showCanceableProgressDialogAndWait(exportService);
+
+        });
+    }
+
+    private void showDialog(List<CopyFilesResultViewModel> data) {
+        Dialog<ButtonType> dlg = new Dialog<>();
```

Comment:
Can you please extract the result dialog to a seperate fxml file, as all the other dialogs.

Reply:
Will do later. All other issues should be resolved now